audb/core/api.py

```diff
@@ -44,15 +44,33 @@ def available(
     databases = []
     for repository in config.REPOSITORIES:
         try:
-            backend = utils.access_backend(repository)
-            if isinstance(backend, audbackend.Artifactory):
-                # avoid backend.ls('/')
-                # which is very slow on Artifactory
-                # see https://github.com/audeering/audbackend/issues/132
-                for p in backend._repo.path:
-                    name = p.name
-                    try:
-                        for version in [str(x).split("/")[-1] for x in p / "db"]:
+            backend_interface = repository()
```

Comment:
This really took me a while to grasp: you are iterating over `config.REPOSITORIES`, so `repository` is  an instance of `Repository` that is defined in `audb` as of now. 

Therefore when doing `repository()` you are going into `__call__`. 
In other words, I found it hard to avoid thinking that this is a constructor.

I ended up thinking about the repository class as a mixture of a kind of data class (a container for the host) and a factory pattern (`__call__` that is returning the interface).

Reply:
Yeah, I think choosing `__call__()` was maybe not the best option.
The main goal of the repository class is to support to configure repositories by using names to refer to backends, e.g. we can write `file-system` or `artifactory` in the config file. This is then converted to the actual backend interface class.

I selected `__call()__` to return the backend instead of the alternative of using the property `backend`, as we have already the attribute `backend`, that stores the backend name.
Any suggestion, how we could improve on this?

Reply:
Not sure whether I can improve this - after all these are conventions.
I mean the `__call__` method creates an object of another class (a backend interface).
This is what I called the "factory" aspect alluding to the factory design pattern. 

The other thing that you are doing is maintaining an associative array of backends and a register method to add to this array. 

All this resembles to the ` factory design pattern`. This is often implemented by having an abstract factory method, that then is implemented by subclassing. However the backend classes themselves, e.g. `audbackend.backend.Artifactory` exist in an  entirely different repository.  This would make it awkward again to have the `Repository` class sit in `audb`

So the only thing that I can possible imagine would be to indeed rename `__call__` - and give it a self-explanatory name, e.g.  `create_audbackend_interface` or whatever.

Reply:
Good idea.
I renamed `__call__()` to `create_backend_interface()` and added am extensive docstring, see the screenshot in the description. I also added a new `test_repository()` file, to explicitly test the behavior of `audb.Repository`.
Please have a look if the docstring and the tests are ok, or if I should extend them.

Reply:
Test are passing. One might add in the docstring to `class Repository` that the class can be used to create a backend interface instance. Still this is good to be resolved.

Reply:
Good idea, I extended the docstring accordingly:

![image](https://github.com/audeering/audb/assets/173624/f5e51e32-5650-4047-b559-2f8dbd8bba8d)